### PR TITLE
feat: add neoclide/coc.nvim integration mini-howto

### PR DESCRIPTION
**Pull request by jjo**

### CODE REVIEW

File: editor/vim/coc-settings.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+    "languageserver": {
+        "jsonnet": {
+        "command": "jsonnet-language-server",
+        "args": ["-t"],
+        "rootPatterns": [".git/"],
+        "filetypes": ["jsonnet"]
```
Comment (zzehring) [inline editor/vim/coc-settings.json:7]: If I'm reading structure right, these should be indented a bit. Also, I think filetypes should include `libsonnet` as well.

Edit: I do see the filetypes here matches `jsonnet-language-server.vim` config. (cc @julienduchesne) Maybe we change this in that file too? I haven't messed around in vim/emacs with the lsp, so maybe it's doing something behind the scenes that I'm not privy to.

File: editor/vim/README.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+## jsonnet-language-server integration for vim
+
+The LSP integration will depend on the vim plugin you're using:
+
+* `mattn/vim-lsp-settings`:
+  * Follow new LSP instalation from <https://github.com/mattn/vim-lsp-settings>
+  * LSP settings file: [jsonnet-language-server.vim](jsonnet-language-server.vim)
+* `neoclide/coc.nvim`:
+  * Inside vim, run: `:CocConfig` (to edit `~/.vim/coc-settings.json`)
+  * Copy [coc-settings.json](coc-settings.json) content
```
Comment (zzehring) [inline editor/vim/README.md:10]: I'm thinking we want to mention here that the args in `coc-settings.json` and `jsonnet-language-server.vim` are adjustable (i.e. they could adjust it JPATH, tanka enablement, linting diagnostics, etc).

**Review by zzehring — COMMENTED** [review]
Thanks for the contribution @jjo! A couple small comments. I haven't messed with the lang server in vim/emacs, so I'll leave the approval to @julienduchesne

**Comment by julienduchesne** [comment]
I know absolutely nothing of vim. Perhaps @Duologic can review this

**Comment by Duologic** [comment]
I've switched to Neovim which has LSP built-in, never used coc.vim. Let me drop a message on the vim slack channel for reviewers.

File: editor/vim/coc-settings.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
+    "languageserver": {
+        "jsonnet": {
+        "command": "jsonnet-language-server",
+        "args": ["-t"],
+        "rootPatterns": [".git/"],
```
Comment (chaudum) [inline editor/vim/coc-settings.json:6]: I would suggest to also add `jsonnetfile.json` as root directory indicator.

File: editor/vim/coc-settings.json (hunk 1 above)
Comment (chaudum) [reply editor/vim/coc-settings.json:7]: +1 for adding `libsonnet` as filetype.

File: editor/vim/coc-settings.json (hunk 3 above)
Comment (jjo) [reply editor/vim/coc-settings.json:6]: +1 doing

File: editor/vim/coc-settings.json (hunk 1 above)
Comment (jjo) [reply editor/vim/coc-settings.json:7]: thanks!, +1 to both, will fix

File: editor/vim/README.md (hunk 2 above)
Comment (jjo) [reply editor/vim/README.md:10]: +1 improving that README stanza

**Comment by jjo** [comment]
Addressed PR feedback at 5b2d291, PTAL -- `golangci-lint` is failing (on unrelated changes re: this PR).

**Comment by julienduchesne** [comment]
> Addressed PR feedback at [5b2d291](https://github.com/grafana/jsonnet-language-server/commit/5b2d291aeca8ae201608edf2457cad4d71c9658f), PTAL -- `golangci-lint` is failing (on unrelated changes re: this PR).

To fix CI, feel free to approve + merge + rebase on https://github.com/grafana/jsonnet-language-server/pull/41

**Comment by jjo** [comment]
> > Addressed PR feedback at [5b2d291](https://github.com/grafana/jsonnet-language-server/commit/5b2d291aeca8ae201608edf2457cad4d71c9658f), PTAL -- `golangci-lint` is failing (on unrelated changes re: this PR).
> 
> I've fixed CI (#41), you can rebase on main

Thanks, done.

**Review by julienduchesne — APPROVED** [review]

**Pull request merged**